tags.go

```diff
@@ -200,3 +206,66 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	// Get underlying resource type for type checking
+	switch resource := resourceType.(type) {
+	case *Domain:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *InfrastructureResource:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Repository:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Service:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *System:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Team:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *User:
+		tagConnection, err = resource.GetTags(client, nil)
+	default:
+		return assignedTags, fmt.Errorf("not a taggable resource type: %s" + string(resourceType.ResourceType()))
+	}
+	if err != nil {
+		return assignedTags, err
+	}
+	if tagConnection == nil {
+		return assignedTags, fmt.Errorf("no tags found on %s with id '%s'", string(resourceType.ResourceType()), resourceType.ResourceId())
+	}
+
+	// delete tags found in resource but not listed in tagsWanted
+	for _, tag := range tagConnection.Nodes {
+		if !slices.Contains(tagsWanted, tag.Flatten()) {
+			if err := client.DeleteTag(tag.Id); err != nil {
+				return assignedTags, err
+			}
+		}
+	}
+
+	// format tags listed in Terraform config but not found in service
+	tagInput := map[string]string{}
+	for _, tag := range tagsWanted {
+		parts := strings.Split(tag, ":")
+		if len(parts) != 2 {
+			return assignedTags, fmt.Errorf("[%s] invalid tag, should be in format 'key:value' (only a single colon between the key and value, no spaces or special characters)", tag)
+		}
+		key := parts[0]
+		value := parts[1]
+		tagInput[key] = value
+	}
+	// assign tags listed in Terraform config but not found in service
+	assignedTags, err = client.AssignTags(string(resourceType.ResourceId()), tagInput)
+	if err != nil {
+		return assignedTags, err
```

Comment:
Same comment about error joining

Reply:
There's a single err here. What needs joining?

Reply:
You are right, `AssignTags` only returns one error since it's just a single API call - resolving this